packages/themes/build.js

```diff
@@ -0,0 +1,243 @@
+  { category: 'global', subitem: { not: ['seed'] } },
+  { category: 'spacing' }
+];
+
+const conditionsTw = [
+  { category: 'theme', type: 'light' },
+  { category: 'global', subitem: { not: ['light', 'dark', 'seed'] } },
+  { category: 'spacing' }
+];
+
+const conditionsJs = [
+  { category: 'theme' },
+  {
+    category: 'global',
+    type: { not: ['spacing'] },
+    item: { not: ['mode'] },
+    subitem: { not: ['seed'] }
+  }
+];
+
+const matchesCondition = (token, condition) => {
+  return Object.entries(condition).every(([key, value]) => {
+    if (typeof value === 'object' && value.not) {
+      return !value.not.includes(token.attributes[key]);
+    }
+    return token.attributes[key] === value;
+  });
+};
+
+const tokenMatchesAnyCondition = (token, conditions) => {
+  return conditions.some((condition) => matchesCondition(token, condition));
+};
+
+const createMatcher = (conditions) => (token) => tokenMatchesAnyCondition(token, conditions);
+
+// REGISTER THE CUSTOM FILTERS USING OUR MATCHING CONDITIONS
+
+StyleDictionary.registerFilter({
+  name: 'cssGlobalFilter',
+  matcher: (token) => createMatcher(conditionsCssGlobal)(token)
+});
+
+StyleDictionary.registerFilter({
+  name: 'cssLightThemeFilter',
+  matcher: (token) => createMatcher(conditionsCssLightTheme)(token)
+});
+
+StyleDictionary.registerFilter({
+  name: 'cssDarkThemeFilter',
+  matcher: (token) => createMatcher(conditionsCssDarkTheme)(token)
+});
+
+StyleDictionary.registerFilter({
+  name: 'twFilter',
+  matcher: (token) => createMatcher(conditionsTw)(token)
+});
+
+StyleDictionary.registerFilter({
+  name: 'jsFilter',
+  matcher: (token) => createMatcher(conditionsJs)(token)
+});
+
+/**
+ * Custom format that handle reference in css variables
+ */
+const transformString = (str, replace = 'theme', regex = /.*-color/) => {
+  // Use a regular expression to replace everything before regex with replace
+  return str.replace(regex, replace);
+};
+
+StyleDictionary.registerFormat({
+  name: 'css/variables',
+  formatter({ dictionary }) {
+    return `${this.options.selector} {
+          ${dictionary.allTokens
+            .map((token) => {
+              let originalName = token.name;
+
+              let themedName =
+                token.path[0] === 'global'
+                  ? originalName
+                  : transformString(originalName, 'theme-', /.*-color-/);
+
+              if (dictionary.usesReference(token.original.value)) {
+                const reference = dictionary.getReferences(token.original.value);
+                const referenceName =
+                  reference[0].path[0] === 'global'
+                    ? reference[0].name
+                    : transformString(reference[0].name);
+                return `  --${themedName}: var(--${referenceName}, ${token.value});`;
+              }
+
+              return `  --${themedName}: ${token.value};`;
+            })
+            .join('\n')}
+        }`;
+  }
+});
+
```

Comment:
There is too much nesting in this template string - I'd extract out the anonymous function:

```suggestion

const formatCSSVariable = (token) => {
    let originalName = token.name;

    let themedName =
      token.path[0] === 'global'
        ? originalName
        : transformString(originalName, 'theme-', /.*-color-/);

    if (dictionary.usesReference(token.original.value)) {
      const reference = dictionary.getReferences(token.original.value);
      const referenceName =
        reference[0].path[0] === 'global'
          ? reference[0].name
          : transformString(reference[0].name);
      return `  --${themedName}: var(--${referenceName}, ${token.value});`;
    }

    return `  --${themedName}: ${token.value};`;
}

StyleDictionary.registerFormat({
  name: 'css/variables',
  formatter({ dictionary }) {
    return `${this.options.selector} {
          ${dictionary.allTokens.map(formatCSSVariable).join('\n')}
        }`;
  }
});

```